dspy/clients/lm.py

```diff
@@ -33,6 +33,7 @@ def __init__(
         launch_kwargs: Optional[Dict[str, Any]] = None,
         callbacks: Optional[List[BaseCallback]] = None,
         num_retries: int = 3,
+        suppress_debug_info: bool = True,
```

Comment:
I'd remove this since it doesn't actually control things, i.e. users can't do dspy.configure() and get suppress to turn off

Reply:
after dealing with this one way or another feel free to merge :D